pkg/rebuild/maven/strategy.go

```diff
@@ -61,6 +61,8 @@ func (b *MavenBuild) GenerateFor(t rebuild.Target, be rebuild.BuildEnv) (rebuild
 	return workflow.GenerateFor(t, be)
 }
 
+const gradleVersion = "8.14.3"
```

Comment:
we could consider moving the version hard-coding into the inference itself. that might be nice since rerunning the inference from the same code location would produce the same strategy+script whereas if we bump the version here, the same strategy at the same code location would generate a different script.

In fact, since we can detect when the gradlew is absent in the inference itself, could we go one step further and move this to a property like "ManualGradleVersion" which would conditionally add the deps and build steps to do the build? Another benefit of that approach would be that we would avoid runtime conditionals e.g. "if [ -f gradlew ]"

Reply:
I thought of this before but I did not want to have `GradleBuild` and `GradlewBuild` but perhaps I can have a property like you suggested.

Reply:
I added a property `NeedsSystemGradle` as I wanted the default behaviour to always install `./gradlew`.

Reply:
We have inference like this now
```
{
  "gradle_build": {
    "repo": "https://github.com/tink-crypto/tink-java",
    "ref": "52a599de062bb551c1bf2b6f3e44709e243618be",
    "dir": ".",
    "jdk_version": "11",
    "system_gradle": "8.14.3"
  }
}
```

Reply:
nice lg